diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/generation/WarningsGeneration.kt

```diff
@@ -0,0 +1,55 @@
+package org.cqfn.diktat.ruleset.generation
+
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.TypeSpec
+import java.io.File
+
+
+private const val PATH_TO_WARNINGS = "src/main/kotlin/org/cqfn/diktat/ruleset/constants/Warnings.kt"
+
+private const val AUTO_GENERATION_COMMENT =
+        " This document was auto generated, please dont modify it\n" +
+        " This document contains all enum properties from Warnings.kt as Strings"
+
+fun main(){
+    val bufferedReader = File(getPathToWarnings()).bufferedReader()
+    val lineList = mutableListOf<String>()
+
+    bufferedReader.useLines { lines ->
+        val stringSeq = lines.filter { it.contains(Regex("([A-Z_]{5,}\\()")) }
+
+        stringSeq.forEach {
+            val enumConstName = it.
+                    replaceAfter("(", "").
+                    dropLast(1).
+                    trim()
+            lineList.add(enumConstName) }
+    }
+
+    val propertyList = mutableListOf<PropertySpec>()
+
+    lineList.forEach {
+        val prop = PropertySpec.builder(it, String::class)
+                .addModifiers(KModifier.CONST)
+                .initializer("\"$it\"")
+                .build()
+
+        propertyList.add(prop)
+    }
```

Comment:
```suggestion
    val propertyList = lineList.map {
         PropertySpec.builder(it, String::class)
                .addModifiers(KModifier.CONST)
                .initializer("\"$it\"")
                .build()
    }
```
Generally in kotlin, since you have distinct mutable and immutable collections, best practices suggest to use immutable whenever possible. For example, here you don't need to populate a list inside `forEach`, because you can simply transform every element using `map`.